dotnet/src/Connectors/Connectors.OpenAI/Core/OpenAIFunction.cs

```diff
@@ -129,47 +142,55 @@ internal OpenAIFunction(
     /// <returns>A <see cref="ChatTool"/> containing all the function information.</returns>
     public ChatTool ToFunctionDefinition()
```

Comment:
strict mode should be a parameter here instead. And we should remove all the properties that have been added. Things should then fall in place to remove the parameter from the invoke prompt async method.